perf/perftool.py

```diff
@@ -71,7 +71,7 @@ def test(self):
         '''
         count = 0
         # Built in perf test
-        for string in process.run("perf test").stderr.splitlines():
+        for string in process.run("perf test", ignore_status=True).stderr.splitlines():
```

Comment:
@PraveenPenguin 
1. why are you splitting the lines multiple times ?
2. Are we fine in failing fast here ? If so, the for loop is unnecessary

Reply:
@narasimhan-v no it takes time to run test , and it goes one by one test case as example this is output
```
11:26:41 DEBUG| [stderr]  1: vmlinux symtab matches kallsyms            : Skip
11:26:41 DEBUG| [stderr]  2: Detect openat syscall event                : Ok
11:26:41 DEBUG| [stderr]  3: Detect openat syscall event on all cpus    : Ok
11:26:41 DEBUG| [stderr]  4: Read samples using the mmap interface      : Ok
11:26:41 DEBUG| [stderr]  5: Test data source output                    : Ok
11:26:41 DEBUG| [stderr]  6: Parse event definition strings             : Ok
11:26:41 DEBUG| [stderr]  7: Simple expression parser                   : Ok
11:26:41 DEBUG| [stderr]  8: PERF_RECORD_* events & perf_sample fields  : Ok
11:26:41 DEBUG| [stderr]  9: Parse perf pmu format                      : Ok
11:26:41 DEBUG| [stderr] 10: DSO data read                              : Ok
11:26:41 DEBUG| [stderr] 11: DSO data cache                             : Ok
11:26:41 DEBUG| [stderr] 12: DSO data reopen                            : FAILED!
11:26:41 DEBUG| [stderr] 13: Roundtrip evsel->name                      : Ok
11:26:41 DEBUG| [stderr] 14: Parse sched tracepoints fields             : Ok
11:26:41 DEBUG| [stderr] 15: syscalls:sys_enter_openat event fields     : Ok
11:26:41 DEBUG| [stderr] 16: Setup struct perf_event_attr               : FAILED!
11:26:41 DEBUG| [stderr] 17: Match and link multiple hists              : Ok
11:26:41 DEBUG| [stderr] 18: 'import perf' in python                    : Ok
11:26:41 DEBUG| [stderr] 22: Watchpoint                                 :
11:26:41 DEBUG| [stderr] 22.1: Read Only Watchpoint                     : FAILED!
11:26:41 DEBUG| [stderr] 22.2: Write Only Watchpoint                    : FAILED!
11:26:41 DEBUG| [stderr] 22.3: Read / Write Watchpoint                  : FAILED!
11:26:41 DEBUG| [stderr] 22.4: Modify Watchpoint                        : FAILED!
11:26:41 DEBUG| [stderr] 23: Number of exit events of a simple workload : Ok

```

Reply:
@PraveenPenguin so, the for loop is not needed here. You can parse it as a one whole string.